ranger/core/progress_tracker.py

```diff
@@ -37,4 +45,5 @@ def get_status_text(self):
 
 class FileProgressTracker(ProgressTracker):
     def get_status_text(self):
-        return '{:3.2f}%'.format(self.percent)
+        average_rate = human_readable(self.average_rate, '') + '/s'
+        return '{:3.2f}% {:>7}'.format(self.percent, average_rate)
```

Comment:
```suggestion
        rate = human_readable(self.average_rate, '')
        return '{:3.2f}% {:>5}/s'.format(self.percent, rate)
```

Are 5 characters even enough?

Reply:
I don't like your suggestion, because in it the contents of `rate` variable happen to have incorrect units if taken out of context (e.g. `60MB`). I'd like to have the status items either preformatted completely on a separate row (this case) or formatted in the template string (e.g. percent). This would be a hybrid between the two.

I will rename the variable to `rate` no problem. I might eventually rename it back to `average_rate` when I decide to implement current rate so there will be two rates to distinguish between.

>Are 5 characters even enough?

It should be, the `human_readable()` seems to never produce more than 4 digits + separator + unit(1 character). Since empty separator is used, this turns out to be 5 characters. See the function docstring for example outputs. Anyways, I am unhappy with having the character count tightly coupled to the `human_readable()` output format. The `.format()` should not know how many characters does each statistic need. However, I have no other solution to prevent jumping of the whole text when the rate changes (e.g. from `40M/s` to `39.9M/s`). Maybe fixed lengths for all of them?

Reply:
> I'd like to have the status items either preformatted completely on a separate row (this case) or formatted in the template string (e.g. percent). This would be a hybrid between the two.

I don't understand that argument. On one hand, you want the unit in the final format string, and on the other you don't.

> The `.format()` should not know how many characters does each statistic need.

But that's precisely what `.format` is for. I don't see a reasonbale way to do it differently.

> 4 digits + separator + unit(1 character)

I see, that makes sense. I didn't check the docstring or source for `human_readable`, so I wasn't sure about its result. Since the rate is given in bytes, afaik, that should also be part of the unit and become `40MB/s`.

Reply:
> I don't understand that argument. On one hand, you want the unit in the final format string, and on the other you don't.

I am sorry for confusing explanation. Alright, ignore the two categories (preformatted completely, formatted in the template string), I realized they are not the actual argument. In your suggestion: if I would pause the code before executing the `format()`, the immediate value would lack the `/s` and therefore it wouldn't be a valid rate representation. It would be a representation of data size. If the variable was named something like `data_per_second` then the variable name *would* actually correspond to its content and I would be fine with it. I hope its clear now.

> But that's precisely what .format is for. I don't see a reasonbale way to do it differently, unless human_readable gets a parameter for column width or number precision.

Should I refactor `human_readable()` then? I don't think I want to go that far right now though. 

> Since the rate is given in bytes, afaik, that should also be part of the unit and become 40MB/s.

That's another reason to refactor the function. How should this be kept compatible with the tree view (`gui/widgets/browsercolumn.py`) which (for some reason) uses single character file size units ('K', 'M', 'G' etc.)? Something like a `abbreviate_units` function parameter?

Reply:
>  It would be a representation of data size.

Yes, that's a reason I can get behind.

> Should I refactor `human_readable()` then?

Definitely not my decision, but I would refrain from doing so. I don't see much value in that, unless there are different places where it would be useful. I believe having the column width in the using format string is fine.

> That's another reason to refactor the function.

Eh, no, you could just add the "B" for display in the format string The `human_readable` function doesn't need to know the unit to do its job, you can just append it afterwards, if desired.

Reply:
@FichteFoll, what do you mean by "implicit format spec numbering?" The empty `{}`'s for positional formatting arguments work perfectly well in python 2.7.

Reply:
Yes, that is what I meant. I definitely remember that not working for me in 2.6, so maybe it was added for 2.7. However, since ranger still supports 2.6, this is still relevant.

Unfortunately, I don't have a 2.6 interpreter at hand to verify.

Reply:
Ah, yep, you're right. Nvm, me, I'm just over here doing nothing of interest...

Reply:
> Yes, that's a reason I can get behind. Ideally we'd put the human_readable call in the same line as .format, but that might be a bit too long.

Ok, I will use your suggestion but with a different variable name.

> Eh, no, you could just add the "B" for display in the format string The human_readable function doesn't need to know the unit to do its job, you can just append it afterwards, if desired.

That would work for all but bytes. In case the size is below 1K the `human_readable()` adds a `B` as a unit. This would result in `BB/s` in these cases. I don't see a way to correct the units without refactoring the function or creating a wrapper(s) for it. I will investigate this in this PR thread but separated from this discussion.

> However, since ranger still supports 2.6, this is still relevant.

Alright, will be testing all changes in Python 2.6 from now on. I will probably use a Vagrant box, because I don't want to mess up my system by downgrading packages. This box looks promising: https://app.vagrantup.com/pcbadger/boxes/centos6-64-python2-6.

Reply:
Checking docs don't mention "Since 2.7 or 3.x" is usually enough, and running `make test` with a python 2, 2.7 is fine. Python 3 doesn't support the pylint version we're still using.
I'm working on updating to newer pylint but it involves manually implementing some checkers to prevent python 2 incompatibilities from sneaking in.